plugins/inputs/directory_monitor/directory_monitor.go

```diff
@@ -285,24 +282,17 @@ func (monitor *DirectoryMonitor) parseFile(parser parsers.Parser, reader io.Read
 	return nil
 }
 
-func (monitor *DirectoryMonitor) parseLine(parser parsers.Parser, line []byte, firstLine bool) ([]telegraf.Metric, error) {
+func (monitor *DirectoryMonitor) parseLine(parser parsers.Parser, line []byte) ([]telegraf.Metric, error) {
 	switch parser.(type) {
 	case *csv.Parser:
-		// The CSV parser parses headers in Parse and skips them in ParseLine.
-		if firstLine {
-			return parser.Parse(line)
-		}
-
-		m, err := parser.ParseLine(string(line))
+		m, err := parser.Parse(line)
 		if err != nil {
+			if errors.Is(err, io.EOF) {
+				return []telegraf.Metric{}, nil
```

Comment:
Maybe doing this would be even better
```suggestion
				return nil, nil
```